core/domain/blog_statistics_services.py

```diff
@@ -0,0 +1,1094 @@
+    blog_post_reads_stats.reads_by_date[current_month_year][current_day] += 1
+    blog_post_reads_stats.reads_by_month[current_year][current_month] += 1
+
+    author_blog_post_reads_stats.reads_by_hour[current_date][current_hour] += 1
+    author_blog_post_reads_stats.reads_by_date[
+        current_month_year][current_day] += 1
+    author_blog_post_reads_stats.reads_by_month[
+        current_year][current_month] += 1
+
+    blog_post_reads_stats.repack_stats()
+    author_blog_post_reads_stats.repack_stats()
+
+    save_blog_post_reads_stats_model(blog_post_reads_stats)
+    save_author_blog_post_reads_stats_model(author_blog_post_reads_stats)
+
+
+@transaction_services.run_in_transaction_wrapper
+def _update_reading_time_stats_transactional(
+    blog_post_id: str,
+    author_id: str,
+    time_taken_to_read_post: int
+) -> None:
+    """Updates Blog Post Reading Time Model and Author Blog Post Reading Time
+    Model. The model GET and PUT must be done in a transaction to avoid loss of
+    updates that come in rapid succession.
+
+    Args:
+        blog_post_id: str. ID of the blog post.
+        author_id: str. ID of the author of the blog post.
+        time_taken_to_read_post: int. The time taken to read the blog post.
+
+    Raises:
+        Exception. BlogPostReadingTimeModel and AuthorBlogPostReadingTimeModel
+            does not exist.
+    """
+    blog_post_reading_time_stats = (
+        get_blog_post_reading_time_stats_by_id(blog_post_id)
+    )
+    author_blog_post_reading_time_stats = (
+        get_author_blog_posts_reading_time_stats_by_id(author_id)
+    )
+    _increment_reading_time_bucket_count(
+        blog_post_reading_time_stats, time_taken_to_read_post)
+    _increment_reading_time_bucket_count(
+        author_blog_post_reading_time_stats, time_taken_to_read_post)
+
+    save_blog_post_reading_time_model(blog_post_reading_time_stats)
+    save_author_blog_posts_aggregated_reading_time_model(
+        author_blog_post_reading_time_stats
+    )
+
+
+@overload
+def _increment_reading_time_bucket_count(
+    stats: blog_statistics_domain.BlogPostReadingTime,
+    time_taken: int
+) -> None: ...
+
+
+@overload
+def _increment_reading_time_bucket_count(
+    stats: blog_statistics_domain.AuthorBlogPostsReadingTime,
+    time_taken: int
+) -> None: ...
+
+
+def _increment_reading_time_bucket_count(
+    stats: Union[
+        blog_statistics_domain.BlogPostReadingTime,
+        blog_statistics_domain.AuthorBlogPostsReadingTime
+    ],
+    time_taken: int
+) -> None:
+    """Increments the value of the time bucket corresponding to time taken by
+    user to read the blog post.
+
+    Args:
+        stats: BlogPostReadsAggregatedStats|AuthorBlogPostReadsAggregatedStats.
+            The stats domain object for which weekly reads are to be generated.
+        time_taken: int. Time taken by the user to read the blog post.
+    """
+    if time_taken == 0:
```

Comment:
We can create a list with stats keys `zero_to_one_min`, `one_to_two_min` and so on as values and their indices as keys for `time_taken` and then increment the required key by mapping with the time_taken value. It will remove too many if-else statements here